1.2/Source/LHM/HediffComp_LuciferiumHeal.cs

```diff
@@ -143,6 +144,31 @@ private void ReduceAgeOfHumanlike()
             }
         }
 
+        private void ReduceAgeOfNonHumanlike()
+        {
+            int lifeStage = Pawn.ageTracker.CurLifeStageIndex;
+            long startOfThirdStage = (long)(Pawn.RaceProps.lifeStageAges[2].minAge * GenDate.TicksPerYear);
+            long diffFromOptimalAge = Pawn.ageTracker.AgeBiologicalTicks - startOfThirdStage;
+
+            Pawn.ageTracker.AgeBiologicalTicks -= diffFromOptimalAge / 600;
+            if (Settings.Get().ShowAgingMessages)
+            {
+                Pawn.ageTracker.AgeBiologicalTicks.TicksToPeriod(out int biologicalYears, out int biologicalQuadrums, out int biologicalDays, out float biologicalHours);
+                string ageBefore = "AgeBiological".Translate(biologicalYears, biologicalQuadrums, biologicalDays);
+                Pawn.ageTracker.AgeBiologicalTicks -= diffFromOptimalAge / 600;
+
+                Pawn.ageTracker.AgeBiologicalTicks.TicksToPeriod(out biologicalYears, out biologicalQuadrums, out biologicalDays, out biologicalHours);
+                string ageAfter = "AgeBiological".Translate(biologicalYears, biologicalQuadrums, biologicalDays);
+                Messages.Message("MessageAgeReduced".Translate(
+                        Pawn.LabelShort,
+                        ageBefore,
+                        ageAfter
+                    ),
+                    MessageTypeDefOf.PositiveEvent, true
+                );
+            }
+        }
```

Comment:
You should feel free to use the MIT-licensed CryoRegenesis de-aging code. It's much more simple and efficient.

Here's all you'd need:

```
            isTargetAge = pawn.ageTracker.AgeBiologicalTicks <= ((GenDate.TicksPerYear * this.targetAge) + rate);
            if (pawn.ageTracker.AgeBiologicalTicks > GenDate.TicksPerYear * targetAge) {
                pawn.ageTracker.AgeBiologicalTicks = Math.Max(pawn.ageTracker.AgeBiologicalTicks - rate, GenDate.TicksPerYear * targetAge);
            }